File: 100-Days-Series/Day 24/PerfectNumber.py

# Function for check if number is perfect number or not.
def getIsPerfectNumber(num):
    if num<=1:
        return False
    sum1=0
    for i in range(1,(num+2)//2,1):
        if num%i==0:
            sum1+=i
    return sum1==num

# Let's Start here
num=28
# Call the function and display result.
if getIsPerfectNumber(num):
    print(num," is a perfect number!")
else:
    print(num," is not a perfect number!")

File: 100-Days-Series/Day 24/PerfectNumber.java
public class PerfectNumber {
    // Function for check number is perfect or not.
    public static boolean isPerfect(int num) {
        if (num <= 1) return false;

        int sum = 0;
        for (int i = 1; i <= num / 2; i++) {
            if (num % i == 0) {
                sum += i;
            }
        }
        return sum == num;
    }
    //Main Method
    public static void main(String[] args) {
        // Let's Start here
        java.util.Scanner sc = new java.util.Scanner(System.in);
        System.out.print("Enter a number: ");
        int number = sc.nextInt();
        sc.close();
        // Call the function and display result.
        if (isPerfect(number)) {
            System.out.println(number + " is a Perfect Number.");
        } else {
            System.out.println(number + " is NOT a Perfect Number.");
        }
    }
}
